lib/recipes-data/src/lib/extract-recipes.test.ts

```diff
@@ -375,9 +378,123 @@ describe("extractRecipeData", () => {
+              typeData: {
+                aspectRatio: "5:3",
+                width: 500,
+                height: 300
+              }
+            },
+            {
+              type: AssetType.IMAGE,
+              mimeType: "image/jpeg",
+              file: "http://media.guim.co.uk/58c32a98ae4463b5129bf717b1b2312d8ffc0d45/0_318_5512_3308/master/5512.jpg",
+              typeData: {
+                aspectRatio: "5:3",
+                width: 5512,
+                height: 3308,
+                isMaster: true
+              }
+            }
+          ],
+          imageTypeData: {
+            caption: "Thomasina Miers’ soba noodles with crisp rainbow vegetables and a spicy sesame seed dressing.",
+            copyright: "LOUISE HAGGER",
+            displayCredit: true,
+            credit: "Photograph: Louise Hagger for the Guardian. Food styling: Emily Kydd. Prop styling: Jennifer Kay",
+            source: "Louise Hagger for the Guardian. Food styling: Emily Kydd. Prop styling: Jennifer Kay",
+            alt: "Thomasina Miers’ soba noodles with crisp rainbow vegetables and a spicy sesame seed dressing.",
+            mediaId: "58c32a98ae4463b5129bf717b1b2312d8ffc0d45",
+            mediaApiUri: "https://api.media.gutools.co.uk/images/58c32a98ae4463b5129bf717b1b2312d8ffc0d45",
+            suppliersReference: "Soba noodles with crisp rainbow vegetables and a spicy sesame se",
+            imageType: "Photograph"
+          }
+        }
+      ]
+    }
+    const activeSponsorships: Sponsorship[] = [
+      {
+        sponsorshipType: SponsorshipType.SPONSORED,
+        sponsorName: "theguardian.org",
+        sponsorLogo: "https://static.theguardian.com/commercial/sponsor/22/Feb/2024/f459c58b-6553-486d-939a-5f23fd935f78-Guardian.orglogos-for badge.png",
+        sponsorLink: "https://theguardian.org/",
+        aboutLink: "https://www.theguardian.com/global-development/2010/sep/14/about-this-site",
+        sponsorLogoDimensions: {
+          width: 280,
+          height: 180
+        },
+        highContrastSponsorLogo: "https://static.theguardian.com/commercial/sponsor/22/Feb/2024/3d8e52dc-1d0b-4f95-8cd1-48a674e1309d-guardian.org new logo - white version (3).png",
+        highContrastSponsorLogoDimensions: {
+          width: 280,
+          height: 180
+        },
+        validFrom: makeCapiDateTime("2023-09-21T16:18:10Z"),
+        validTo: makeCapiDateTime("2026-08-30T23:00:00Z"),
+      }
+    ]
+    const result = extractRecipeData(canonicalId, block, activeSponsorships)
+    expect(result.length).toEqual(1)
+    expect(result[0]?.recipeUID).toEqual("62ac3f0f98f6495cbefd72c11fac6d1e26390e99")
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-assignment -- this is safe
+    const data = JSON.parse(JSON.parse(JSON.stringify(result))[0]?.jsonBlob)
+    const sponsorsExists = "sponsors" in data
+    expect(sponsorsExists).toBe(true)
```

Comment:
Great start - please can we verify that the fields actually come through correctly, and that there is exactly one sponsor in the data?

Reply:
yes now checking them as well 👍